src/server/graphics/multiplexing_display.cpp

```diff
@@ -181,25 +184,47 @@ class CompositeDisplayConfiguration : public mg::DisplayConfiguration
         };
         std::vector<OutputInfo> const output_map;
 
+        static auto name_for_output(
+            mg::DisplayConfigurationOutput const& output,
+            std::map<mg::DisplayConfigurationCardId, std::map<mg::DisplayConfigurationOutputType, unsigned>>& output_counts,
+            bool multiple_cards) -> std::string
+        {
+            output_counts[output.card_id][output.type]++;    // The map will default-initialise any as-yet-unseen cards to the empty map
+                                                             // and any as-yet-unseen output types to 0.
+            std::stringstream name;
+            name << mir::output_type_name(static_cast<unsigned>(output.type));
+            if (multiple_cards)
+            {
+                name << "-" << output.card_id.as_value() + 1;
+            }
+            name << "-" << output_counts[output.card_id][output.type];
+            return name.str();
+        }
```

Comment:
This doesn't quite match the behaviour of the existing `name_outputs()` implementations. I think it would avoid problems to match the current behaviour (and would be simple to implement).

The existing behaviour:

Card 0: names are not decorated with a card number
Card n, [n>0] : names are decorated with n

This PR:

Card 0: names are either not decorated with a card number or, if there are multiple cards, decorated with 1
Card n, [n>0] : names are decorated with n+1

I don't see the motivation for making names on card 0 change when another card is added, nor for changing the names on subsequent cards.

Reply:
Agreed, let's try and stick to the names we have.